tensorflow_addons/optimizers/discriminative_layer_training.py

```diff
@@ -21,8 +21,14 @@
 from tensorflow_addons.optimizers import KerasLegacyOptimizer
 from typeguard import typechecked
 
-from keras import backend
-from keras.utils import tf_utils
+try:
+    # New versions of Keras require importing from `keras.src` when
+    # importing internal symbols.
+    from keras.src import backend
```

Comment:
Do we need this line?  Shouldn't `from keras import backend` work since its public API?

Reply:
Below, on line 144, `backend._current_graph()` is called. Since this is an internal symbol, it requires the `keras.src` import. `from keras import backend` will only allow public symbols to be accessed on `backend`, while `from keras.src import backend` allows all symbols from `backend.py` to be accessed.

Reply:
Thanks for the details.  Sounds good.